Source/Assets/PlayFabEditorExtensions/Editor/Scripts/Panels/PlayFabEditorPackages.cs

```diff
@@ -0,0 +1,89 @@
+using UnityEditor;
+using UnityEngine;
+using System;
+using System.Reflection;
+
+namespace PlayFab.PfEditor
+{
+    public class PlayFabEditorPackages : Editor
+    {
+        private const int buttonWidth = 150;
+
+        public static bool IsPubSubInstalled { get { return GetIsPubSubInstalled(); } }
+
+        public static void DrawPackagesMenu()
+        {
+            var labelStyle = new GUIStyle(PlayFabEditorHelper.uiStyle.GetStyle("label"));
+            if(!IsPubSubInstalled)
+            {
+                GUILayout.BeginHorizontal();
+                GUILayout.Label("PersistentSockets: ");
+                if (GUILayout.Button("Install", PlayFabEditorHelper.uiStyle.GetStyle("Button"), GUILayout.MaxWidth(buttonWidth), GUILayout.MinHeight(32)))
+                {
+                    string possibleNewtonsoftPath = "";
+                    if(GetIsNewtonsoftInstalled(out possibleNewtonsoftPath))
+                    {
+                        EditorUtility.DisplayDialog("Newtonsoft is already installed.",
+                            "Please delete your version of Netwonsoft.json.dll in \n\n" + possibleNewtonsoftPath + " \n and retry the install.\n\n compiler conflicts will happen if this stays around", "Continue", "Cancel");
+                    }
+                    else
+                    {
+                        ImportPubSubSDK();
+                    }
+                }
+
+                GUILayout.EndHorizontal();
+            }
+            else
+            {
+                EditorGUILayout.LabelField("PersistentSockets is already Installed", labelStyle, GUILayout.MinWidth(EditorGUIUtility.currentViewWidth));
+            }
+        }
+
+        public static void ImportPubSubSDK()
+        {
+            // TODO: when the PR is checked in, we'll need to use the below url on api.playfab.com 
+            // "https://github.com/PlayFab/UnitySDK/tree/master/Packages/PubSub.unitypackage";
+
+            var tempLink = "https://api.playfab.com/downloads/unity-signalr";
+            PlayFabEditorHttp.MakeDownloadCall(tempLink, (fileName) =>
+            {
+                AssetDatabase.ImportPackage(fileName, false);
+            });
+        }
+
+        public static bool GetIsNewtonsoftInstalled(out string path)
+        {
+            var allAssemblies = AppDomain.CurrentDomain.GetAssemblies();
+            foreach (var assembly in allAssemblies)
+            {
+                if (assembly.FullName.Contains("Newtonsoft.Json"))
+                {
+                    path = assembly.Location;
+                    return true;
+                }
+            }
+            path = "N/A";
+            return false;
+        }
+
+        public static bool GetIsPubSubInstalled()
+        {
+            var allAssemblies = AppDomain.CurrentDomain.GetAssemblies();
```

Comment:
We need to do this action a number of times with a variety of types.
We should put this function as a utility somewhere, which caches the information in a way that makes lookups by name faster, and we'd only have to call GetAssemblies and Foreach/foreach once.
As we need to do this more and more, it's going to be a severe performance bottleneck when we're constantly checking for these types existing in loops.

Reply:
Maybe skip it for THIS checkin, but it's gonna be a problem "soon".